CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# Contributing
+
+We would ❤️ for you to contribute to Appwrite and help make it better! As a contributor, here are the guidelines we would like you to follow:
+
+## Code of Conduct
+
+Help us keep Appwrite open and inclusive. Please read and follow our [Code of Conduct](/CODE_OF_CONDUCT.md).
+
+## Getting Started
+
+Install using composer:
+```bash
+composer require appwrite/php-runtimes
+```
+
+```php
+<?php
+
+require_once '../vendor/autoload.php';
+
+use Appwrite\Runtimes\Runtimes;
+
+$runtimes = new Runtime();
+
+// returns all supported cloud function runtimes
+Runtimes::getAll();
+```
+
+## Adding a new Runtime
+
+After the Docker image is created, this must be added to the main class of this library.
+
+References to this must be added to the constructor of `src/Runtimes/Runtimes.php`.
+
+Example:
+
+```php
+$dotnet = new Runtime('dotnet', '.NET');
+$dotnet->addVersion('5.0', 'mcr.microsoft.com/dotnet/runtime:5.0-alpine', 'appwrite/env-dotnet-5.0:1.0.0', [System::X86, System::ARM]);
+$runtimes[] = $dotnet;
+```
+
+## Function Runtimes Checklist
+
+The following checklist aims to ensure that a function runtime gets added successfully
+
+- [ ] Implement the runtime in [open-runtimes/open-runtimes](https://github.com/open-runtimes/open-runtimes) (you can find the tutorial [here](https://github.com/open-runtimes/open-runtimes/blob/main/docs/add-runtime.md))
+  - [ ] Prepare the Readme for your new runtime. Make sure to mention the following details
+    - [ ] Docker base image name + version
+    - [ ] HTTP server library name
+    - [ ] Any extra steps needed to get dependencies running (for e.g., `package.json`)
+    - [ ] Copy the rest of the Readme from another existing runtime
+  - [ ] Write the runtime
+    - [ ] Initialize a web server
+      - Use a library that supports `async/await`, also smaller/simpler libraries are preferred as this is a primitive HTTP server
+      - [ ] Set Port 3000
+      - [ ] Bind IP 0.0.0.0
+      - On each POST Request
+        - [ ] Check that the `x-internal-challenge header` matches the `INTERNAL_RUNTIME_KEY` environment variable
+        - [ ] Decode the executor's JSON POST request
+          - [ ] Make sure to have the right default values for Request body fields ([example](https://github.com/open-runtimes/open-runtimes/blob/main/runtimes/node-16.0/server.js#L14-L18))
+           - [ ]  env: empty object
+           - [ ]  headers: empty object
+           - [ ]  payload: empty string
+    - [ ] Create Request Class
+      - [ ] Fields
+        - [ ] env
+        - [ ] payload
+        - [ ] headers
+    - [ ] Create Response Class
+      - [ ] Functions
+        - [ ] send(string data, int statusCode)
+        - [ ] json(object data, int statusCode)
```

Comment:
Somewhere we should note that the runtime should set the correct content-type header for json, and use the plaintext contant-type for send.

Reply:
Are we mandatorily setting this?
I was looking at the `[dotnet-6.0](https://github.com/open-runtimes/open-runtimes/blob/a277d1b36ebdfb9b0d4fd81866733f8df21b53d1/runtimes/dotnet-6.0/src/Program.cs#L30)` runtime and we just keep the `application/json` type.

Aside from using a string for the `send` function and Dictionaries/Maps/objects that can be JSONified for the `json` function, I'm not sure if we're manually adding the `Content-Type` header in most cases.

Reply:
```suggestion
        - [ ] json(object data, int statusCode)
        - When returning a response, the `Content-Type` header for the `send()` function is `text/plain` and for the `json` function is `application/json`
```

@PineappleIOnic does this look alright?

Reply:
LGTM 👍